nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,516 @@
+                    "as an independent message. If this value is populated, the message will be sent as a reply to the message whose timestamp is equal to the given value.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TEXT = new PropertyDescriptor.Builder()
+            .name("text")
+            .displayName("Text")
+            .description("Text of the Slack message to send. Only required if no attachment has been specified and 'Upload File' has been set to 'No'.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor INITIAL_COMMENT = new PropertyDescriptor.Builder()
+            .name("upload-text")
+            .displayName("Upload Text")
+            .description("Text of the Slack message to include with a public upload. Only used if 'Publish Mode' has been set to 'Upload Only' and 'Upload Channel' is specified.")
+            .dependsOn(PUBLISH_STRATEGY, UPLOAD_ONLY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor FILE_TITLE = new PropertyDescriptor.Builder()
+            .name("file-title")
+            .displayName("File Title")
+            .description("Title of the file displayed in the Slack message." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'.")
+            .dependsOn(PUBLISH_STRATEGY, UPLOAD_ONLY, UPLOAD_ATTACH, UPLOAD_LINK)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor FILE_NAME = new PropertyDescriptor.Builder()
+            .name("file-name")
+            .displayName("File Name")
+            .description("Name of the file to be uploaded." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'." +
+                    " If the property evaluated to null or empty string, then the file name will be set to 'file' in the Slack message.")
+            .dependsOn(PUBLISH_STRATEGY, UPLOAD_ONLY, UPLOAD_ATTACH, UPLOAD_LINK)
+            .defaultValue("${" + CoreAttributes.FILENAME.key() + "}")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
```

Comment:
Specifying the filename is not really necessary, since there is already a well-known attribute that specifies the filename. If user was to use a different filename, they should change the filename attribute. While having the extra property doesn't necessarily hurt anything, it does lead to more properties that the user is forced to consider, so best to avoid.